### ISSUE TITLE
Play separate video and audio together?

### ISSUE THREAD

**Issue by rpw98** [wang-bin/QtAV#1074]
I have an online video source that has separate video and audio streams, both are mp4 files. How to play them together? Is it possible?

Thank you!
Rp

**Comment by wang-bin** [comment]
yes. you can use another media that contains audio streams as audio track for player. `AVPlayer.setExternalAudio(url) ` use the first audio track of external media file. Then call `setAudioStream(index)` to change to a new track in the external media if necessary. 
 https://github.com/wang-bin/QtAV/blob/master/src/QtAV/AVPlayer.h#L198

**Comment by rpw98** [comment]
Thank you. I've been experimenting with it in the last week. It plays. But often the video quality get bad and pixellated and sometimes even appears slow-motioned.  A also sometimes video and audio got out-of-sync. If I don't load the audio stream, video plays smoothly. Is this something that can be improved? 

Again, thanks for the help! Aside from the little issues here and there, this toolkit is really essential to me. Keep up the good work!

**Comment by flaviotordini** [comment]
I am too having bad performance when using `AVPlayer.setExternalAudio(url)`. The same video plays fluidly without external audio. Any idea what the cause maybe be?

**Comment by flaviotordini** [comment]
In my case, the bad performance happens with a webm/vp9 video stream and does NOT happen with same video encoded in mp4/h264
